src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,39 @@
 package mate.academy;
 
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private final ConcurrentHashMap<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        CompletableFuture<UserData> completableFuture = new CompletableFuture<>();
+
+        completableFuture.completeAsync(() -> {
+            if (cache.containsKey(userId)) {
+                return cache.get(userId);
+            }
+
+            UserData userData = getUserDataFromBase(userId);
+            cache.put(userId, userData);
+
+            return userData;
+        }, executor);
+        return completableFuture;
```

Comment:
use static method (factory method pattern). 

```suggestion
        return CompletableFuture.supplyAsync(() -> {....}, executors);
```